service/history/transferQueueTaskExecutorBase.go

```diff
@@ -262,20 +166,17 @@ func (t *transferQueueTaskExecutorBase) recordWorkflowClosed(
 	searchAttributes *commonpb.SearchAttributes,
 ) error {
 
-	// Record closing in visibility store
-	namespace := defaultNamespace
-	archiveVisibility := false
-
 	namespaceEntry, err := t.shard.GetNamespaceCache().GetNamespaceByID(namespaceID)
-	if err != nil && !isWorkflowNotExistError(err) {
-		return err
+	if err != nil {
+		if _, ok := err.(*serviceerror.NotFound); !ok {
+			return err
+		}
+		return nil
```

Comment:
you do not have to explicitly do this error checking, task processing will handle this, e.g. missing mutable state

Reply:
oh.. now I see what you mean in previous PR.